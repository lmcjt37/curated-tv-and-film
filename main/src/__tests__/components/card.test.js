import React from 'react';
import ReactDOM from 'react-dom';

import Card from '../../components/card';

import { createRender } from '@material-ui/core/test-utils';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

const itemMock = {
  type: 'tv_show',
  title: 'Mock of Thrones',
  description: 'You know nothing',
  season: 1,
  episode: 1,
  episode_title: 'To kill a mockingbird',
  imdb: 'https://www.lmcjt.com',
  url: 'https://www.lmcjt.com',
  thumbnail: './assets/images/game-of-thrones.jpg',
  genre: ['Action', 'Adventure', 'Drama']
};

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Card {...itemMock} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('snapshot of initial component', () => {
  let render = createRender();
  const component = render(<Card {...itemMock} />);
  expect(component).toMatchSnapshot();
});
