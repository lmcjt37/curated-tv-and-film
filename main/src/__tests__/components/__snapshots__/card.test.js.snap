// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of initial component 1`] = `
<div
  class="MuiPaper-root-142 MuiPaper-elevation1-145 MuiPaper-rounded-143 MuiCard-root-141 SingleCard-card-132"
>
  <div
    class="SingleCard-innerContainer-133"
  >
    <div
      class="MuiCardMedia-root-169 SingleCard-media-134"
      style="background-image:url(\\"./assets/images/game-of-thrones.jpg\\")"
      title="Mock of Thrones"
    />
    <img
      alt="Mock of Thrones"
      class="SingleCard-thumbnail-135"
      src="./assets/images/game-of-thrones.jpg"
    />
    <div
      class="MuiCardContent-root-171"
    >
      <h2
        class="MuiTypography-root-172 MuiTypography-h6-189 MuiTypography-gutterBottom-199"
      >
        Mock of Thrones 
      </h2>
      <div>
        <h3
          class="MuiTypography-root-172 MuiTypography-subtitle1-190 MuiTypography-gutterBottom-199"
        >
          To kill a mockingbird
        </h3>
        <p
          class="MuiTypography-root-172 MuiTypography-body1-181 MuiTypography-colorTextSecondary-205"
        >
          Season: 1 | Episode: 1
        </p>
      </div>
      <div
        class="SingleCard-chips-136"
      >
        <div
          class="MuiChip-root-207 SingleCard-chip-137"
          role="button"
          tabindex="-1"
        >
          <span
            class="MuiChip-label-226"
          >
            Action
          </span>
        </div>
        <div
          class="MuiChip-root-207 SingleCard-chip-137"
          role="button"
          tabindex="-1"
        >
          <span
            class="MuiChip-label-226"
          >
            Adventure
          </span>
        </div>
        <div
          class="MuiChip-root-207 SingleCard-chip-137"
          role="button"
          tabindex="-1"
        >
          <span
            class="MuiChip-label-226"
          >
            Drama
          </span>
        </div>
      </div>
      <p
        class="MuiTypography-root-172 MuiTypography-body1-181"
      >
        You know nothing
      </p>
      <div
        class="MuiCardActions-root-232"
      >
        <a
          aria-label="IMDB"
          class="MuiButtonBase-root-244 MuiFab-root-234 MuiFab-extended-238 MuiFab-primary-236 SingleCard-fab-138 MuiCardActions-action-233"
          href="https://www.lmcjt.com"
          rel="noopener noreferrer"
          role="button"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiFab-label-235"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-247 SingleCard-localPlayIcon-140"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M20 12c0-1.1.9-2 2-2V6c0-1.1-.9-2-2-2H4c-1.1 0-1.99.9-1.99 2v4c1.1 0 1.99.9 1.99 2s-.89 2-2 2v4c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2v-4c-1.1 0-2-.9-2-2zm-4.42 4.8L12 14.5l-3.58 2.3 1.08-4.12-3.29-2.69 4.24-.25L12 5.8l1.54 3.95 4.24.25-3.29 2.69 1.09 4.11z"
              />
            </svg>
            IMDB
          </span>
        </a>
        <a
          aria-label="View Scene"
          class="MuiButtonBase-root-244 MuiFab-root-234 MuiFab-extended-238 MuiFab-primary-236 SingleCard-fab-138 MuiCardActions-action-233"
          href="https://www.lmcjt.com"
          rel="noopener noreferrer"
          role="button"
          tabindex="0"
          target="_blank"
        >
          <span
            class="MuiFab-label-235"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-247 SingleCard-playCircleOutlineIcon-139"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M10 16.5l6-4.5-6-4.5v9zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
              />
            </svg>
            View Scene
          </span>
        </a>
      </div>
    </div>
  </div>
</div>
`;
